proof/crefine/RISCV64/TcbAcc_C.thy

```diff
@@ -120,13 +120,12 @@ lemma register_from_H_less:
   by (cases hr, simp_all add: "StrictC'_register_defs")
 
 lemma register_from_H_sless:
-  "register_from_H hr <s 35"
+  "UCAST(8 \<rightarrow> 32 signed) (register_from_H hr) <s 0x23"
```

Comment:
To avoid ate least one magic number in this one, I'd just use `ucast` without length annotation, but give the number on the right a type (32 signed from what it looks like). From that Isabelle, should be able to infer the concrete `ucast` type directly.

Reply:
I think `UCAST(_ -> 32 signed)` would also avoid one magic number, but I can't tell which style I prefer.

But perhaps even better would be to make some type synonyms:
```
type_synonym register_idx_len = 8
type_synonym register_idx = "register_idx_len word"
type_synonum int_literal_len = "32 signed"
```
Then you can write `UCAST(register_idx_len -> int_lteral_len)`, and also `register_idx` in the type of `register_from_H`.

Reply:
That'd be even better, yes

Reply:
I guess somewhere near the top of `Wellformed_C.thy` would be a reasonable place for these type synonyms?